cmake/FindSUNDIALS.cmake

```diff
@@ -122,7 +122,7 @@ if (SUNDIALS_INCLUDE_DIR)
 endif()
 
 if("${SUNDIALS_VERSION_MAJOR}" LESS 4)
-  message(WARNING "SUNDIALS versions <4 are depricated and will not be supported in the next release")
+  message(FATAL_ERROR "SUNDIALS versions <4 are not supported")
 endif()
```

Comment:
We could actually remove this bit entirely and change our call to `find_package` in `cmake/SetupBOUTThirdParty.cmake` to `find_package(SUNDIALS 4 REQUIRED)`